modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -258,6 +237,44 @@ object CrossSources {
       WithBuildRequirements(BuildRequirements(), _)
     )
 
+    val allPathsWithDirectivesByScope: Map[Scope, Seq[(os.Path, DirectivesPositions)]] =
+      (pathsWithDirectivePositions ++ inMemoryWithDirectivePositions)
+        .flatMap { (withBuildRequirements, directivesPositions) =>
+          val scope = withBuildRequirements.scopedValue(Scope.Main).scope
+          val path: os.Path = withBuildRequirements.value match
+            case im: Sources.InMemory =>
+              im.originalPath match
+                case Right((_, p: os.Path)) => p
+                case _                      => inputs.workspace / im.generatedRelPath
+            case (p: os.Path, _) => p
+          directivesPositions.map((path, scope, _))
+        }
+        .groupBy((_, scope, _) => scope)
+        .map { (scope, entries) =>
+          scope -> entries.map((path, _, directivesPositions) => path -> directivesPositions)
+        }
+    val anyScopeHasMultipleSourcesWithDirectives =
+      Scope.all.exists(allPathsWithDirectivesByScope.get(_).map(_.length).getOrElse(0) > 1)
+    val shouldSuppressWarning =
```

Comment:
It should be `lazy val` to avoid initialising this variable when spread directive warning is disabled.